unit_test/fuzzing/test_requester/test_spdm_requester_get_event_types/get_event_types.c

```diff
@@ -0,0 +1,146 @@
+                                                size_t *response_size,
+                                                void **response,
+                                                uint64_t timeout)
+{
+    libspdm_test_context_t *spdm_test_context;
+    uint8_t *spdm_response;
+    size_t spdm_response_size;
+    size_t test_message_header_size;
+    uint32_t session_id;
+    libspdm_session_info_t *session_info;
+
+    spdm_test_context = libspdm_get_test_context();
+
+    session_id = m_session_id;
+    session_info = libspdm_get_session_info_via_session_id(spdm_context, session_id);
+
+    test_message_header_size = LIBSPDM_TEST_TRANSPORT_HEADER_SIZE;
+    spdm_response = (void *)((uint8_t *)temp_buf + test_message_header_size);
+    spdm_response_size = spdm_test_context->test_buffer_size;
+    if (spdm_response_size > sizeof(temp_buf) - test_message_header_size - LIBSPDM_TEST_ALIGNMENT) {
+        spdm_response_size = sizeof(temp_buf) - test_message_header_size - LIBSPDM_TEST_ALIGNMENT;
+    }
+    libspdm_copy_mem((uint8_t *)temp_buf + test_message_header_size,
+                     sizeof(temp_buf) - test_message_header_size,
+                     spdm_test_context->test_buffer,
+                     spdm_response_size);
+
+    libspdm_transport_test_encode_message(spdm_context, &session_id,
+                                          false, false, spdm_response_size,
+                                          spdm_response, response_size, response);
+
+    /* Workaround: Use single context to encode message and then decode message. */
+    ((libspdm_secured_message_context_t *)(session_info->secured_message_context))->
+    application_secret.response_data_sequence_number--;
+
+    return LIBSPDM_STATUS_SUCCESS;
+}
+
+static void libspdm_test_requester_get_event_types_case1(void **state)
+{
+    libspdm_test_context_t *spdm_test_context;
+    libspdm_context_t *spdm_context;
+    libspdm_session_info_t *session_info;
+    uint32_t session_id;
+    uint8_t event_group_count;
+    uint8_t supported_event_groups_list[LIBSPDM_MAX_SPDM_MSG_SIZE];
+    uint32_t supported_event_groups_list_len;
+
+    spdm_test_context = *state;
+    spdm_context = spdm_test_context->spdm_context;
+
+    spdm_context->connection_info.version = SPDM_MESSAGE_VERSION_13 <<
+                                            SPDM_VERSION_NUMBER_SHIFT_BIT;
+    spdm_context->connection_info.connection_state = LIBSPDM_CONNECTION_STATE_NEGOTIATED;
+
+    spdm_context->connection_info.capability.flags =
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_EVENT_CAP |
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_ENCRYPT_CAP |
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MAC_CAP |
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_KEY_EX_CAP;
+
+    spdm_context->local_context.capability.flags =
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_ENCRYPT_CAP |
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MAC_CAP |
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_KEY_EX_CAP;
```

Comment:
Here these three CAP should be changed to as following for requester, right? 
`SPDM_GET_CAPABILITIES_REQUEST_FLAGS_ENCRYPT_CAP`
`SPDM_GET_CAPABILITIES_REQUEST_FLAGS_MAC_CAP`
`SPDM_GET_CAPABILITIES_REQUEST_FLAGS_KEY_EX_CAP`